pkg/storegateway/bucket.go

```diff
@@ -1187,7 +1204,18 @@ func (s *BucketStore) recordSeriesCallResult(safeStats *safeQueryStats) {
 	s.metrics.cachedPostingsCompressedSizeBytes.Add(float64(stats.cachedPostingsCompressedSizeSum))
 	s.metrics.seriesHashCacheRequests.Add(float64(stats.seriesHashCacheRequests))
 	s.metrics.seriesHashCacheHits.Add(float64(stats.seriesHashCacheHits))
-	s.metrics.expandPostingsDuration.Observe(stats.expandedPostingsDuration.Seconds())
+
+	// Track the streaming store-gateway preloading effectiveness metrics only if the request had
+	// more than 1 batch. If the request only had 1 batch, then preloading is not triggered at all.
+	if stats.streamingSeriesBatchCount > 1 {
+		s.metrics.streamingSeriesBatchPreloadingLoadDuration.Observe(stats.streamingSeriesBatchLoadDuration.Seconds())
+		s.metrics.streamingSeriesBatchPreloadingWaitDuration.Observe(stats.streamingSeriesWaitBatchLoadedDuration.Seconds())
```

Comment:
I have a similar comment about the histogram resolution - we're measuring the total duration of preloading as a single measurement.

This means we mix potential differences in the preloading efficiency for different blocks.

I think this is more important for `ExpandedPostings` since there I expect to get different loading times for different blocks, but for preloading series and chunks it may be negligible.

Reply:
We use these metrics to track the preloading efficiency. The efficiency makes sense at a per `Series()` request level. We want to know how much time of `Series()` request is reduced thanks to preloading.

Reply:
i see. that makes sense to me. We lose the ability to see whether changes we make make a difference in e.g. loading chunks specifically (regardless of the queries), but I don't think we were using these particular metrics to determine that anyways. 

higher-level stats seem good